Port to AHK v2

greener-hayden opened the pull request:
I have ported the AHK code so its able to run on AHK v2. The code has been stable but there could be outliers or problems that i haven’t encountered yet so be warned. The only thing missing from this conversion is the validation of whether a method exists. So, if the user presses a key that hasn’t been updated, it will result in an error so keep that in mind.
In addition, I am currently working on integrating a plugin with Flow-Launcher. There is also a possibility of integrating with Powertoys Run, but for now, my focus is on Flow since it seems like a more agile system since i can build the plugin with python.

I also have plans to enhance the code options by allowing users to set parameter values for the methods. Improving the ability of the plugin on stream deck. I also have plans to add descriptions/flavour text to functions. Which should improve the aesthetics of hooking HotkeylessAHK to other things.

itslightmind commented:
Wanted to include my test results, 13700k, 3080, 32GB, Win11
| Approach                | Delay |
| ----------------------- | :------- |
| Key combinations        | ~ 04 ms  |
| Single AHK scripts      | ~ 76 ms | 
| *Hotkeyless AutoHotkey* | ~ 36 ms |

greener-hayden commented:
Could you please look over this pull @sebinside ?
I've now added support for parameters. Currently can do strings, integers, floats, boolean, arrays, objects, and nulls. However, I couldn’t figure out how to pass mathematical operations, so it’s not a complete one-to-one conversion of the method(params*). Nevertheless, it covers about 90% of the functionality and 100% of what I needed.

I hope that the port is at the quality that it could be merged.

![image](https://github.com/sebinside/HotkeylessAHK/assets/36151712/683f10d7-158e-4876-a728-6114ee69fc15)

sebinside commented:
Thank you for this giant PR! Looks like a great improvement of the overall code base. However, I will need to find quite some time to review (and first, to understand) the changes and additions. In my current extremely tight schedule, this will probably take some time, I'm sorry. Still, thank you very much, this PR will eventually be merged :)

greener-hayden commented:
## Multiple Function Class Support

This update introduces support for multiple function classes, allowing for better organization and separation of functions based on their purpose or context. For example, you can now have classes like `WorkFunctions`, `PersonalFunctions`, etc., in addition to the main `CustomFunctions` class.

The changes made include:

- Added support for multiple function classes in the `GetAvailableFunctions()` function. The `functionClasses` array can be expanded to include additional function classes as needed.
- Modified the `register` endpoint to handle multiple function classes. It now extracts the function class and function name from the registered list and stores them in the `functionClassMap`.
- Updated the `send` endpoint to find the corresponding function class for a given function name using the `findFunctionClass` method. If a matching function class is found, the command is sent with the appropriate class prefix.

sebinside reviewed (approved):
One year later, I finally found time to review this PR. First of all, thank you very much for this comprehensive overhaul and update to v2. I only have minor issues and remaining tasks for this update, then I will release a new AHK version with your changes. Thanks again!
